source/MaterialXView/Viewer.cpp

```diff
@@ -2452,7 +2562,11 @@ mx::ImagePtr Viewer::getShadowMap()
             blurSamplingProperties.uaddressMode = mx::ImageSamplingProperties::AddressMode::CLAMP;
             blurSamplingProperties.vaddressMode = mx::ImageSamplingProperties::AddressMode::CLAMP;
             blurSamplingProperties.filterType = mx::ImageSamplingProperties::FilterType::CLOSEST;
-            for (unsigned int i = 0; i < _shadowSoftness; i++)
+
+            // This is expensive to compute every frame for a turntable render.
+            // Compute this however if a capture is requisted.
+            unsigned int softness = (_turntableEnabled && !_captureRequested) ? 0 : _shadowSoftness;
```

Comment:
I didn't examine this but this code is quite slow and makes the updates very laggie.

Reply:
@kwokcb In my experience, it's more common for turntable systems to rotate the _camera_ around a fixed mesh, rather than rotating the mesh itself.  This would eliminate the problem you ran into here, as shadows would no longer need to be recomputed on each frame of the turntable render.  Would that potentially be a better overall approach to implementing this new feature?

Reply:
Both approaches to rendering turntables are certainly valid, and there are cases where a user would prefer to keep the camera static and rotate the mesh.  But to robustly implement this form of turntable rendering in MaterialXView, we'd need to address the underlying performance limitation by moving shadow rendering entirely to the GPU, rather than implementing it as a hybrid GPU/CPU pass.

That shadow optimization would be valuable in any case, but I wouldn't recommend trying to work around it by disabling individual shadow passes during turntable rendering, since this effectively reduces the visual quality of turntable renders below the quality of our standard viewport renders.

Reply:
I can remove this to not mess with quality vs frame rate. BTW other options slow frame rate (like upping the number of environment samples). "Interactive degradation" options could be added if that's a route worth pursuing.

Perhaps we could try this option first, as a camera based turntable has different characteristics.

Camera rotation also means the view of the background environment changes so to me is more of a camera "fly-by".
Currently the camera is free to move about so a rotating mesh can be evaluated from different angles.
Finally I think you may end up with some unexpected camera viewpoints if someone does transform the camera before running a turntable. I guess you could fix the "look at" point to the origin, or scene center but rotations on the camera could end up with strange results. 

Does this sound reasonable?

Reply:
Good points about mesh vs camera rotation, and I think you're right to focus on mesh rotation for now.  In that case, I would recommend keeping your change as minimal as possible, and leaving performance optimizations as a future improvement.  My sense is that optimizing shadow rendering would make the most difference, removing the CPU step that copies final pixels into a texture and making this a pure GPU operation.

Reply:
If I have some time, I can take a look at how to create a all hardware "pipeline" but probably not until later.
I know we don't want to formalize MaterialX as a "pipeline" specifier but I may be tempted to try it out in this context.
To solve this would also to allow MaterialX to generate shaders for "multi-pass" web rendering (nothing is enabled now).

Reply:
@jstone-lucasfilm. I ran this with on a Mac M1 machine and there is only a very small perceivable lag for the shadow filtering option turned on. Can probably log something once I figure out what the characteristics are.